components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/store/UserStoreBasedIdentityDataStore.java

```diff
@@ -72,6 +72,9 @@ public void store(UserIdentityClaimsDO userIdentityDTO, UserStoreManager userSto
             try {
                 // Check if the user store is read only. If it is read only and still uses user store based data
                 // store then log a warn.
+                if (!userStoreManager.isExistingUser(username)){
```

Comment:
Don't we check isExistingUser, in user core functions before invoking the listner?